### PR TITLE
Docs - Minor Note to assist future readers

### PR DESCRIPTION
**Pull request by CharlKlein**
## Problem Statement

Minor updates to templating docs to assist future readers, caught me out today.

## Related Issue

None

## Proposed Changes

Minor documentation update

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [ ] My changes have reasonable test coverage
- [ ] All tests pass with `make test`
- [ ] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: docs/guides/templating.md [hunk 1]
```diff
@@ -2,48 +2,54 @@
...
+
+    Example: service-account-token -> serviceAccountToken
 
 ## Helm
 
 When installing ExternalSecrets via `helm`, the template must be escaped so that `helm` will not try to render it. The most straightforward way to accomplish this would be to use backticks ([raw string constants](https://pkg.go.dev/text/template#hdr-Examples)):
 
 ```yaml
-{% include 'helm-template-v2-escape-sequence.yaml' %}
+{ % include 'helm-template-v2-escape-sequence.yaml' % }
```
Comment (moolen) [inline docs/guides/templating.md:16]: these whitespaces unfortunately break the template 
2023-11-02T19:18:59,977502962+01:00

File: docs/guides/templating.md [hunk 2]
```diff
@@ -2,48 +2,54 @@
 
 With External Secrets Operator you can transform the data from the external secret provider before it is stored as `Kind=Secret`. You can do this with the `Spec.Target.Template`. Each data value is interpreted as a [golang template](https://golang.org/pkg/text/template/).
 
+!!! note
+
+    **.'spec.data.secretkey'** cannot contain **'-'** (dashes), consider using camelcase when defining the secret keys.
```
Comment (moolen) [inline docs/guides/templating.md:7]: Good suggestion, this definitely will help people! Though technically you can use dashes, it's just a pain in the ass to work with :sweat_smile: 

Something along the lines of:
```
{{  index . "service-account-token" }}
```

**Review by moolen — COMMENTED** [review]
Hey @CharlKlein :wave:, how is it going? :smiley:

Thanks for you PR, appreciated! Left a comment below.

File: docs/guides/templating.md (hunk 1 above)
Comment (CharlKlein) [reply docs/guides/templating.md]: Im keeping well, Hope the Same. 

That was my auto-formatter being a a little over zealous 🤣

File: docs/guides/templating.md (hunk 2 above)
Comment (CharlKlein) [reply docs/guides/templating.md]: Adjusted a bit to add a note around using the index function

**Comment by CharlKlein** [comment]
Some minor tweaks following the your notes 😄

**Review by moolen — APPROVED** [review]
LGTM, thank you! :bow:

**Pull request merged**